### PR TITLE
Fix warning about comparison between signed and unsigned integers

### PR DESCRIPTION
**Pull request by esteve**
Fixes warning

@dirk-thomas @tfoote @wjwwood

### CODE REVIEW

**Comment by wjwwood** [comment]
+1

File: rclcpp/include/rclcpp/node_impl.hpp [hunk 1]
```diff
@@ -309,13 +309,14 @@ Node::list_parameters(
...
 
-  // TODO: define parameter separator, use "." for now
+  // TODO(esteve): define parameter separator, use "." for now
   for (auto & kv : parameters_) {
     if (std::any_of(prefixes.cbegin(), prefixes.cend(), [&kv, &depth](const std::string & prefix) {
       if (kv.first.find(prefix + ".") == 0) {
         size_t length = prefix.length();
         std::string substr = kv.first.substr(length);
-        return std::count(substr.begin(), substr.end(), '.') < depth;
+        // Cast as unsigned integer to avoid warning
```
Comment (dirk-thomas) [inline rclcpp/include/rclcpp/node_impl.hpp]: I would remove the comment since it doesn't add much value (just my personal opinion).

**Comment by dirk-thomas** [comment]
+1

**Comment by tfoote** [comment]
+1

File: rclcpp/include/rclcpp/node_impl.hpp (hunk 1 above)
Comment (esteve) [inline rclcpp/include/rclcpp/node_impl.hpp]: I added it because I thought it wasn't obvious why I `static_cast`'d `std::count`

**Pull request merged**